lib/apiservers/engine/backends/container/viccontainer.go

```diff
@@ -27,8 +33,77 @@ type VicContainer struct {
 	HostConfig  *containertypes.HostConfig
 }
 
+// NewVicContainer returns a reference to a new VicContainer
 func NewVicContainer() *VicContainer {
 	return &VicContainer{
 		Config: &containertypes.Config{},
 	}
 }
+
+// SetConfigOptions is a place to add necessary container configuration
+// values that were not explicitly supplied by the user
+func (c *VicContainer) SetConfigOptions(config *containertypes.Config) {
+	// Overwrite or append the image's config from the CLI with the metadata from the image's
+	// layer metadata where appropriate
+	if len(config.Cmd) == 0 {
+		config.Cmd = c.Config.Cmd
+	}
+	if config.WorkingDir == "" {
+		config.WorkingDir = c.Config.WorkingDir
+	}
+	if len(config.Entrypoint) == 0 {
+		config.Entrypoint = c.Config.Entrypoint
+	}
+
+	// set up environment
+	setEnv(config, c.Config)
+}
+
+func setEnv(config, imageConfig *containertypes.Config) {
+	// Set TERM to xterm if tty is set
+	if config.Tty {
+		config.Env = append(config.Env, "TERM=xterm")
```

Comment:
Yep, that's what docker does. It should be OK as long as our defaults comes first as the last one (in that case the cli provided one) wins :)